### PR TITLE
Refactor fuse options

### PR DESCRIPTION
**Pull request by dbw9580**
### What changes are proposed in this pull request?

Refactor command line options and parsing logic in Fuse.

### Why are the changes needed?

To make the control flow more extensible.

### Does this PR introduce any user facing changes?

No.

### CODE REVIEW

File: dora/integration/fuse/src/main/java/alluxio/fuse/AlluxioFuse.java [hunk 1]
```diff
@@ -133,18 +76,33 @@ protected AlluxioFuse() {
...
+      InstancedConfiguration globalConf = Configuration.modifiableGlobal();
+      globalConf.merge(configFromCli.getProperties());
+      Source metadataCacheSizeSource =
+          globalConf.getSource(PropertyKey.USER_METADATA_CACHE_MAX_SIZE);
+      if (metadataCacheSizeSource == Source.DEFAULT
+          || metadataCacheSizeSource == Source.CLUSTER_DEFAULT) {
+        globalConf.set(PropertyKey.USER_METADATA_CACHE_MAX_SIZE, 20000, Source.RUNTIME);
+        LOG.info("Set default metadata cache size to 20,000 entries "
+            + "with around 40MB memory consumption for FUSE");
+      }
```
Comment (huanghua78) [inline dora/integration/fuse/src/main/java/alluxio/fuse/AlluxioFuse.java:102]: I don't understand why it is hard-coded to overwrite a default configuration.
If such, we can change the default value for this config key.

I know this is from the original code. But please talk to @LuQQiu to fix this.

File: dora/integration/fuse/src/main/java/alluxio/fuse/AlluxioFuse.java [hunk 2]
```diff
@@ -297,105 +255,78 @@ public static FuseUmountable launchFuse(FuseUmountable fuseFs,
...
+        .ifPresent(updateCheckEnabled -> {
+          conf.set(FUSE_UPDATE_CHECK_ENABLED, updateCheckEnabled, Source.RUNTIME);
+        });
+    cli.getRootUfsUri()
+        .ifPresent(ufsRootUri -> {
+          conf.set(FuseOptions.FUSE_UFS_ROOT, ufsRootUri.toString(), Source.RUNTIME);
+          // Disable connections between FUSE and server
+          conf.set(PropertyKey.USER_METRICS_COLLECTION_ENABLED, false, Source.RUNTIME);
+          conf.set(PropertyKey.USER_UPDATE_FILE_ACCESSTIME_DISABLED, true, Source.RUNTIME);
+        });
```
Comment (huanghua78) [inline dora/integration/fuse/src/main/java/alluxio/fuse/AlluxioFuse.java:277]: How to make sure the `MountPoint` and `RootUfsUri` are present?
They must exist. This is mandatory.

**Review by huanghua78 — COMMENTED** [review]
No objections from me.
Please ask @LuQQiu or someone else to review this.

File: dora/integration/fuse/src/main/java/alluxio/fuse/AlluxioFuse.java (hunk 2 above)
Comment (dbw9580) [reply dora/integration/fuse/src/main/java/alluxio/fuse/AlluxioFuse.java:277]: if absent, the default values for `PropertyKey.FUSE_MOUNT_POINT` and `DORA_CLIENT_UFS_ROOT` will be used. which defaults to `/mnt/alluxio-fuse` and `<worker directory>/underFSStorage` respectively.

File: dora/integration/fuse/src/main/java/alluxio/fuse/options/FuseCliOptions.java [hunk 3]
```diff
@@ -0,0 +1,126 @@
...
+  @Parameter(
+      names = {"--update-check"},
+      description = "Enables or disables the FUSE version update check. "
+          + "Disabled by default when connecting to Alluxio system cache or Dora cache. "
+          + "Enabled by default when connecting an under storage directly.",
+      arity = 0,
+      required = false
+  )
+  @Nullable
+  protected Boolean mUpdateCheck = null;
```
Comment (LuQQiu) [inline dora/integration/fuse/src/main/java/alluxio/fuse/options/FuseCliOptions.java:65]: this is better to be not shown to users

**Review by LuQQiu — APPROVED** [review]
LGTM, thanks!

File: dora/integration/fuse/src/main/java/alluxio/fuse/options/FuseCliOptions.java (hunk 3 above)
Comment (dbw9580) [reply dora/integration/fuse/src/main/java/alluxio/fuse/options/FuseCliOptions.java:66]: changed to a hidden option

**Comment by ssz1997** [comment]
This change breaks the backward compatibility of k8s helm chart, and no one mentioned anything about the change we need to do on k8s side. 
And the title "refactor" is misleading. Refactor supposedly should not change user behavior. And yes, this PR changes the user behavior.

**Pull request merged**